Index: zxing-1.0.0/CMakeLists.txt
===================================================================
--- zxing-1.0.0.orig/CMakeLists.txt
+++ zxing-1.0.0/CMakeLists.txt
@@ -6,25 +6,6 @@ option(BUILD_TESTING "Enable generation
 
 set(CMAKE_LIBRARY_PATH /opt/local/lib ${CMAKE_LIBRARY_PATH})
 
-# Check for polluted source tree.
-if(EXISTS ${CMAKE_SOURCE_DIR}/CMakeCache.txt OR
-    EXISTS ${CMAKE_SOURCE_DIR}/CMakeFiles)
-    message(FATAL_ERROR
-        "Source directory is polluted:"
-        "\n  * remove CMakeCache.txt"
-        "\n  * remove CMakeFiles directory")
-endif()
-
-# Suppress in-source builds.
-if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
-    message(FATAL_ERROR
-        "CMake generation is not allowed within the source directory:"
-        "\n  * mkdir build"
-        "\n  * cd build"
-        "\n  * Unix-like: cmake -G \"Unix Makefiles\" .."
-        "\n  * Windows: cmake -G \"Visual Studio 10\" ..")
-endif()
-
 # Adjust CMake's module path.
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/FindModules)
 
@@ -57,8 +38,13 @@ if(OpenCV_FOUND)
 endif()
 
 include_directories(core/src)
-add_library(libzxing ${LIBZXING_FILES})
+add_library(libzxing SHARED ${LIBZXING_FILES})
+add_library(libzxing_static STATIC ${LIBZXING_FILES})
 set_target_properties(libzxing PROPERTIES PREFIX "")
+set_target_properties(libzxing_static PROPERTIES OUTPUT_NAME "zxing")
+set_target_properties(libzxing PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+set_target_properties(libzxing_static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
+set_target_properties(libzxing PROPERTIES VERSION 1.0.0 SOVERSION 1)
 
 find_package(Iconv)
 if(ICONV_FOUND)
@@ -87,7 +73,7 @@ file(GLOB_RECURSE ZXING_FILES
 add_executable(zxing ${ZXING_FILES})
 target_link_libraries(zxing libzxing)
 
-install(TARGETS zxing libzxing EXPORT zxing-targets
+install(TARGETS zxing libzxing libzxing_static EXPORT zxing-targets
 	LIBRARY DESTINATION lib
 	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib
