Index: opencv-4.1.1/3rdparty/carotene/CMakeLists.txt
===================================================================
--- opencv-4.1.1.orig/3rdparty/carotene/CMakeLists.txt
+++ opencv-4.1.1/3rdparty/carotene/CMakeLists.txt
@@ -21,7 +21,11 @@ if(CMAKE_COMPILER_IS_GNUCC)
     # - goodFeaturesToTrack 10-20%
     # - cornerHarris 30% for some cases
 
-    set_source_files_properties(${carotene_sources} COMPILE_FLAGS "--param ipcp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "10.0.0")
+        set_source_files_properties(${carotene_sources} COMPILE_FLAGS "--param ipcp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
+    else()
+        set_source_files_properties(${carotene_sources} COMPILE_FLAGS "--param ipa-cp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
+    endif()
 endif()
 
 add_library(carotene_objs OBJECT
Index: opencv-4.1.1/3rdparty/carotene/hal/CMakeLists.txt
===================================================================
--- opencv-4.1.1.orig/3rdparty/carotene/hal/CMakeLists.txt
+++ opencv-4.1.1/3rdparty/carotene/hal/CMakeLists.txt
@@ -90,7 +90,11 @@ set_property(DIRECTORY APPEND PROPERTY C
     #   matchTemplate about 5-10%
     #   goodFeaturesToTrack 10-20%
     #   cornerHarris 30% for some cases
-    set_source_files_properties(impl.cpp $<TARGET_OBJECTS:carotene_objs> COMPILE_FLAGS "--param ipcp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "10.0.0")
+        set_source_files_properties(${carotene_sources} COMPILE_FLAGS "--param ipcp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
+    else()
+        set_source_files_properties(${carotene_sources} COMPILE_FLAGS "--param ipa-cp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
+    endif()
 #    set_source_files_properties(impl.cpp $<TARGET_OBJECTS:carotene_objs> COMPILE_FLAGS "--param ipcp-unit-growth=100000 --param inline-unit-growth=100000 --param large-stack-frame-growth=5000")
   endif()
 
