Index: zbar-0.1.0/Makefile.am
===================================================================
--- zbar-0.1.0.orig/Makefile.am
+++ zbar-0.1.0/Makefile.am
@@ -62,17 +62,18 @@ EXTRA_DIST += perl/MANIFEST perl/README
     perl/t/Processor.t perl/t/Scanner.t perl/t/pod.t perl/t/pod-coverage.t
 
 if WIN32
+SUFFIXES = .rc
 dist_doc_DATA += README.windows
 pkgdata_DATA = $(srcdir)/examples/barcode.png \
     $(srcdir)/examples/scan_image.cpp $(srcdir)/examples/scan_image.vcproj
 
-%-rc.o: %.rc
+.o: .rc
 	$(RC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) -o $@ $<
-%-rc.lo: %.rc
+	$(AM_CPPFLAGS) $(CPPFLAGS) -o $@-rc $< $*.rc
+.lo: .rc
 	$(LIBTOOL) --tag=RC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=compile $(RC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) -o $@ $<
+	$(AM_CPPFLAGS) $(CPPFLAGS) -o $@-rc $< $*.rc
 
 # install to tmp dest and run NSIS to generate installer
 dist-nsis: html-local
Index: zbar-0.1.0/configure.ac
===================================================================
--- zbar-0.1.0.orig/configure.ac
+++ zbar-0.1.0/configure.ac
@@ -3,7 +3,7 @@ AC_PREREQ([2.61])
 AC_INIT([zbar], [0.10], [spadix@users.sourceforge.net])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror foreign subdir-objects std-options dist-bzip2])
+AM_INIT_AUTOMAKE([1.10 -Wall foreign subdir-objects std-options dist-bzip2])
 AC_CONFIG_HEADERS([include/config.h])
 AC_CONFIG_SRCDIR(zbar/scanner.c)
 LT_PREREQ([2.2])
